GW2EIEvtcParser/EIData/ProfHelpers/Revenant/RevenantHelper.cs

```diff
@@ -274,14 +274,67 @@ internal static void ComputeProfessionCombatReplayActors(AbstractPlayer player,
                 var skill = new SkillModeDescriptor(player, Spec.Revenant, ProtectiveSolaceSkill, SkillModeCategory.ProjectileManagement);
                 foreach (EffectEvent effect in protectiveSolaceEffectEvents.Where(x => x.IsAroundDst))
                 {
-                    (long, long) lifespan = effect.ComputeDynamicLifespan(log, 0, effect.Dst, ProtectiveSolaceTabletBuff); // manually disabled or when no more ressources
+                    (long, long) lifespan = effect.ComputeDynamicLifespan(log, 0, effect.Dst, ProtectiveSolaceTabletBuff); // manually disabled or when no more resources
                     var connector = new AgentConnector(effect.Dst);
                     replay.Decorations.Add(new CircleDecoration(240, lifespan, color, 0.2, connector)
                         .UsingSkillMode(skill));
                     replay.Decorations.Add(new IconDecoration(ParserIcons.EffectProtectiveSolace, CombatReplaySkillDefaultSizeInPixel, CombatReplaySkillDefaultSizeInWorld, 0.5f, lifespan, connector)
                         .UsingSkillMode(skill));
                 }
             }
+
+            // Eternity's Requiem
+            if (log.CombatData.TryGetEffectEventsBySrcWithGUID(player.AgentItem, EffectGUIDs.RevenantEternitysRequiemOnPlayer, out IReadOnlyList<EffectEvent> eternitysRequiem))
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Revenant, EternitysRequiem);
+                if (log.CombatData.TryGetEffectEventsBySrcWithGUID(player.AgentItem, EffectGUIDs.RevenantEternitysRequiemHit, out IReadOnlyList<EffectEvent> eternitysRequiemHits))
+                {
+                    foreach (EffectEvent effect in eternitysRequiem)
+                    {
+                        var positions = new List<Point3D>();
+                        foreach (EffectEvent hitEffect in eternitysRequiemHits.Where(x => x.Time >= effect.Time && x.Time <= effect.Time + 2800 && x.Dst == null))
+                        {
+                            positions.Add(hitEffect.Position);
+                            (long, long) lifespanHit = hitEffect.ComputeLifespan(log, 1000);
+                            var connector = new PositionConnector(hitEffect.Position);
+                            replay.Decorations.Add(new CircleDecoration(120, lifespanHit, color, 0.5, connector).UsingFilled(false).UsingSkillMode(skill));
+                        }
+                        (long, long) lifespan = (effect.Time, effect.Time + 2800);
+                        var centralPosition = Point3D.FindCentralPoint(positions);
```

Comment:
I'm quite confused about this, the projectiles look quite random to me, is there a documentation somewhere that indicates that the average of the 9 impacts point is the center of the AoE?
Same remark for citadel bombardment, the impacts don't look to be equally distributed in every direction.

Reply:
Both skills have a radius where the impacts will land, but neither have a displayed aoe.
I was using the player position when the first effects appears, but for eternity's requiem it would cause it to be offset while in movement https://gyazo.com/f9c93cd7bb24e9b43d782e92912fd8d3
Here is how it looks with an extimated central point https://gyazo.com/307784b810dfd63f26d25fdfd82b9248
For both skills the actual damage is coming from the small rings, the large one is just as a general indicator to place the skill icon inside

Reply:
I understand what you are trying to achieve, the part I'm conflicted about is the accuracy of the methodology. How accurate is getting the middle of all impacts compared to the actual center of the AoE?
It feels like it would be better to show only an icon to indicate a general position of where the AoE is and omit the circle for the AoE. Looking at it in game right now, it really does not feel like the average of the impacts is the center of the AoE.

Reply:
> I was using the player position when the first effects appears
And for citadel bombardment it would cause the ring to be at the portal location, so i swapped that to be the portal image and created the new position to locate the skill icon

Reply:
This is how it looks like without the ring if you prefer it https://gyazo.com/c139c81365eea1e6fc9aeb7ea5001778

Reply:
Yes, comment the circle for now. We can add it later if we find a good way to show that the circle is approximative, as of today people would take it as truth which would be incorrect.

Reply:
We could have dashed/dotted lines for approximations, like the line decoration we were talking about the other day, but I don't know how that information would be understood by the user though

Reply:
That depends, dashed lines were proposed to avoid bad overlaps for line decorations and not for approximations, dashed lines being approximation for certain stuff and not others would not work out nicely.